src/datepicker/datepicker-tools.spec.ts

```diff
@@ -305,16 +305,12 @@ describe(`datepicker-tools`, () => {
       months = buildMonths(calendar, new NgbDate(2018, 6, 5), state, i18n, false);
       expect(months).toBe(state.months);
       expect(months.length).toBe(2);
-      // the structures should be swapped:
```

Comment:
As discussed, this if not acceptable, because clearly the:
- behaviour is broken
- you're adapting the test to the broken behaviour

Could fix this:

https://github.com/ng-bootstrap/ng-bootstrap/blob/cca081b13c7bfe36bf09d85fc7c96e2a71dc7aa3/src/datepicker/datepicker-tools.ts#L102

I guess we should make sure that this date also always have the `{day: 1}`...

Second option would be to make the tests always use dates with day 1 → `buildMonths(..., new NgbDate(2018, 7, 1), ...)`, but I think I prefer the first one...